packages/trader/test/index.ts

```diff
@@ -39,25 +33,22 @@ describe("index.js", function() {
 
   let originalEnv: any;
 
-  const pollingDelay = 0; // 0 polling delay creates a serverless bot that yields after one full execution.
-  const errorRetries = 1;
-  const errorRetriesTimeout = 0.1; // 100 milliseconds between preforming retries
   const identifier = "TEST_IDENTIFIER";
   const fundingRateIdentifier = "TEST_FUNDiNG_IDENTIFIER";
 
-  const FinancialContract = getTruffleContract("Perpetual", web3);
-  const Finder = getTruffleContract("Finder", web3);
-  const IdentifierWhitelist = getTruffleContract("IdentifierWhitelist", web3);
-  const AddressWhitelist = getTruffleContract("AddressWhitelist", web3);
-  const MockOracle = getTruffleContract("MockOracle", web3);
-  const Token = getTruffleContract("ExpandedERC20", web3);
-  const SyntheticToken = getTruffleContract("SyntheticToken", web3);
-  const Timer = getTruffleContract("Timer", web3);
-  const UniswapMock = getTruffleContract("UniswapV2Mock", web3);
-  const Store = getTruffleContract("Store", web3);
-  const ConfigStore = getTruffleContract("ConfigStore", web3);
-  const OptimisticOracle = getTruffleContract("OptimisticOracle", web3);
-  const DSProxyFactory = getTruffleContract("DSProxyFactory", web3);
+  const FinancialContract = getTruffleContract("Perpetual", web3 as any);
```

Comment:
changing the import of `web3` also mean that I had to do this type casting. not sure what solutions there are to this.

The error produced was as follows:
![image](https://user-images.githubusercontent.com/12886084/118095147-04cc4f00-b3d0-11eb-87b0-42e1b2205d7c.png)

Reply:
This is really annoying. Seems like it might be caused by different versions of web3 that are being imported here and in core.

I think to make it less annoying to deal with, you should just create a casted version at the top.

```ts
import { web3: uncastWeb3 } from "hardhat";
...

// This is a workaround because the core web3 type and local one are slightly different.
type CoreWeb3 = Parameters<typeof getTruffleContract>[1]; // Grabs the type of the second argument.
const web3 = uncastWeb3 as CoreWeb3; // Note: you could also use any here, but it would make web3 any everywhere in this file.
```

Reply:
damn that's pretty messy. I'll update accordingly

Reply:
this did not work :( I had type issues all over the place when trying this technique.